packages/plugin-rsc/src/plugin.ts

```diff
@@ -1273,15 +1286,83 @@ function generateDynamicImportCode(map: Record<string, string>) {
   return `export default {${code}};\n`
 }
 
-// // https://github.com/vitejs/vite/blob/2a7473cfed96237711cda9f736465c84d442ddef/packages/vite/src/node/plugins/importAnalysisBuild.ts#L222-L230
-function assetsURL(url: string) {
+class RuntimeAsset {
+  runtime: string
+  constructor(value: string) {
+    this.runtime = value
+  }
+}
+
+function parseValueWithRuntime(value: any) {
+  return devalue.uneval(value, (value) => {
+    if (value instanceof RuntimeAsset) {
+      return value.runtime
+    }
+  })
+}
+
+function assetsURL(
+  url: string,
+  {
+    environment,
+    enableRuntimeValue = false,
+  }: {
+    environment: Environment
+    enableRuntimeValue?: boolean
+  },
+) {
+  if (
+    enableRuntimeValue &&
+    environment.mode === 'build' &&
+    typeof config.experimental?.renderBuiltUrl === 'function'
+  ) {
+    const result = config.experimental.renderBuiltUrl(url, {
+      type: 'asset',
+      hostType: 'js',
```

Comment:
Should this check js or css?

Reply:
Here is us telling the developer that uses `renderBuiltUrl` that we will apply the result of `renderBuiltUrl` into a JS file (the manifest file). Other types are being used by Vite for example when `index.html` is used to import a file, and `css` imports, so they can check that to make sure it is safe to use `runtime` or not.